underscore.js

```diff
@@ -521,8 +521,13 @@
   };
 
   // Flatten out an array, either recursively (by default), or just one level.
-  _.flatten = function(array, shallow) {
-    return flatten(array, shallow, false);
+  _.flatten = function(array, depth) {
+    if (!depth && depth !== 0) {
+      depth = Infinity;
+    } else if (depth <= 0) {
+      return _.clone(array);
+    }
+    return flatten(array, depth, false);
```

Comment:
we're internally calling `flatten` with booleans, so I don't know if we want to enforce that at the moment?